src/services/pasteEdits.ts

```diff
@@ -167,3 +161,49 @@ function pasteEdits(
         );
     });
 }
+
+function addImportsForUnresolvedIdentifiers(
+    node: Identifier,
+    updatedFile: SourceFile,
+    originalProgram: Program,
+    updatedProgram: Program,
+    importAdder: ImportAdder,
+    host: LanguageServiceHost,
+    preferences: UserPreferences,
+    formatContext: formatting.FormatContext,
+    cancellationToken: CancellationToken,
+    copiedFrom?: { file: SourceFile; range: TextRange[]; } | undefined,
+    symbolsUsageCopiedFrom?: Map<string, SymbolInfo>,
+) {
+    if (copiedFrom) {

```

Comment:
You technically only need to pass in `symbolsUsageCopiedFrom`. That avoids these two parameters from drifting apart in any intended meaning.